### PR TITLE
Feature/client secret encryption - merge only when ready to migrate existing secrets

### PR DESCRIPTION
**Pull request by lepsalex**
Add support for encrypting client secrets stored in database. #96

### CODE REVIEW

File: src/main/java/org/overture/ego/service/ApplicationService.java [hunk 1]
```diff
@@ -132,9 +132,15 @@ public ClientDetails loadClientByClientId(@NonNull String clientId) throws Clien
...
 
     val application = getByClientId(clientId);
-    if(application == null) throw new ClientRegistrationException("Client ID not found.");
-    if(!application.getStatus().equals(ApplicationStatus.APPROVED.toString())) throw new ClientRegistrationException
+
+    if(application == null) {
+      throw new ClientRegistrationException("Client ID not found.");
+    }
+
+    if(!application.getStatus().equals(ApplicationStatus.APPROVED.toString())) {
```
Comment (andricDu) [inline src/main/java/org/overture/ego/service/ApplicationService.java]: use a static import for `APPROVED`

File: pom.xml [hunk 2]
```diff
@@ -95,6 +95,35 @@
             <version>42.1.4</version>
         </dependency>
 
+        <!-- TestContainers (for testing) -->
+        <dependency>
+            <groupId>org.testcontainers</groupId>
```
[the remark this answers is not in the document]
Comment (lepsalex) [reply pom.xml]: Yup, that's a good one

File: src/main/java/org/overture/ego/security/AESEncryptionService.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  public String encrypt(String text) {
+    try {
+      // Generate random ivLength digit initVector
+      byte[] randomInitVector = ByteBuffer.allocate(ivLength).putInt(random.nextInt(10 ^ ivLength)).array();
+
+      IvParameterSpec iv = new IvParameterSpec(randomInitVector);
+      SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
+
+      Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
```
[the remark this answers is not in the document]
Comment (lepsalex) [reply src/main/java/org/overture/ego/security/AESEncryptionService.java]: why public and why static?

**Comment by andricDu** [comment]
merge conflicts

**Comment by rosibaj** [comment]
Closing this PR as the code-base has deviated. We will work on reimplementing this security fix later.

**Comment by andricDu** [comment]
Sadly closing this. This is a feature we should still implement but the code base has moved. I will not be deleting the branch as it contains useful and valid approaches to this problem.

**Pull request closed without merging**